sklearn/cluster/k_means_.py

```diff
@@ -374,6 +374,10 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
             X += X_mean
         best_centers += X_mean
 
+    if len(set(best_labels)) < n_clusters:
+        warnings.warn("number of distinct clusters found smaller than n_clusters. "
```

Comment:
Do you think a ConvergenceWarning is appropriate?

Reply:
Probably more appropriate than a FitFailedWarning...

Reply:
Better than a FitFailedWarning... :)